File: app/src/main/java/com/example/pictureme/viewmodel/AuthViewModel.java
package com.example.pictureme.viewmodel;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.pictureme.models.User;
import com.example.pictureme.repository.AuthRepository;
import com.google.firebase.auth.FirebaseUser;

public class AuthViewModel extends ViewModel {

    private AuthRepository userRepository;
    private MutableLiveData<FirebaseUser> firebaseUser;
    private MutableLiveData<User> user;
    private MutableLiveData<Boolean> loggedIn;

    public AuthViewModel() {
        userRepository = new AuthRepository();
        firebaseUser = userRepository.getFirebaseUserMutableLiveData();
        user = userRepository.getUserMutableLiveData();
        loggedIn = userRepository.getLoggedInMutableLiveData();
    }

    public LiveData<FirebaseUser> getFirebaseUserLiveData() {
        return firebaseUser;
    }

    public LiveData<User> getUserLiveData() {
        return user;
    }

    public LiveData<Boolean> getLoggedInLiveData() {
        return loggedIn;
    }

    public void register(String email, String password, String username) {
        userRepository.register(email, password, username);
    }

    public void login(String email, String password) {
        userRepository.login(email, password);
    }

    public void signOut() {
        userRepository.signOut();
    }

    public boolean matchingPasswords(String password, String passwordConfirmation) {
        return password.equals(passwordConfirmation);
    }

}

File: app/src/main/java/com/example/pictureme/viewmodels/AuthViewModel.kt
package com.example.pictureme.viewmodels

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.pictureme.data.interfaces.AuthRepository
import com.example.pictureme.data.Response
import com.google.firebase.auth.FirebaseUser
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.launch
import javax.inject.Inject

@HiltViewModel
class AuthViewModel @Inject constructor (
    private val repository: AuthRepository
): ViewModel() {

    private val _authLiveData = MutableLiveData<Response<FirebaseUser>?>()
    val authLiveData: LiveData<Response<FirebaseUser>?> = _authLiveData

    val currentUser: FirebaseUser?
        get() = repository.currentUser

//    init {
//        if (repository.currentUser != null) {
//            _authLiveData.value = Resource.Success(repository.currentUser!!)
//        }
//    }

    fun login(email: String, password: String) = viewModelScope.launch {
        _authLiveData.value = Response.Loading
        val result = repository.login(email, password)
        _authLiveData.value = result
    }

    fun signup(name: String, email: String, password: String) = viewModelScope.launch {
        _authLiveData.value = Response.Loading
        val result = repository.signup(name, email, password)
        _authLiveData.value = result
    }

    fun logout() {
        repository.logout()
        _authLiveData.value = null
    }

}